aviatrix/resource_aviatrix_smart_group_test.go

```diff
@@ -201,6 +201,106 @@ func TestAccAviatrixSmartGroup_reject_bad_k8s_names(t *testing.T) {
 	})
 }
 
+func TestAccAviatrixSmartGroup_s2c(t *testing.T) {
+	skipAcc := os.Getenv("SKIP_SMART_GROUP")
+	if skipAcc == "yes" {
+		t.Skip("Skipping Smart Group test as SKIP_SMART_GROUP is set")
+	}
+	resourceName := "aviatrix_smart_group.s2c"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProvidersVersionValidation,
+		CheckDestroy: testAccSmartGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSmartGroupK8s(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSmartGroupExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", "s2c-test-smart-group"),
+					resource.TestCheckResourceAttrSet(resourceName, "uuid"),
+					resource.TestCheckResourceAttr(resourceName, "selector.0.match_expressions.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "selector.0.match_expressions.0.s2c", "test-mapped-s2c"),
+				),
```

Comment:
I wonder how this should work. `Config: testAccSmartGroupK8s()` creates `aviatrix_smart_group.k8s`.
But the `Check` tests for a resource named `aviatrix_smart_group.s2c`.
Hence the test also fails for me:
```
=== RUN   TestAccAviatrixSmartGroup_s2c
time="2024-08-20T10:46:06+02:00" level=info msg="Getting API token..."
time="2024-08-20T10:46:07+02:00" level=info msg="Parsed Aviatrix login: admin"
time="2024-08-20T10:46:08+02:00" level=info msg="Getting API token..."
time="2024-08-20T10:46:08+02:00" level=info msg="Parsed Aviatrix login: admin"
time="2024-08-20T10:46:09+02:00" level=info msg="Getting API token..."
time="2024-08-20T10:46:09+02:00" level=info msg="Parsed Aviatrix login: admin"
    resource_aviatrix_smart_group_test.go:211: Step 1/2 error: Check failed: Check 1/5 error: no Smart Group resource found: aviatrix_smart_group.s2c
time="2024-08-20T10:46:10+02:00" level=info msg="Getting API token..."
time="2024-08-20T10:46:11+02:00" level=info msg="Parsed Aviatrix login: admin"
time="2024-08-20T10:46:11+02:00" level=info msg="Getting API token..."
time="2024-08-20T10:46:12+02:00" level=info msg="Parsed Aviatrix login: admin"
--- FAIL: TestAccAviatrixSmartGroup_s2c (6.73s)
```
Ideally it should even fail since I probably don't have the correct version of the controller running.